pkg/urbit/jets/d/by_uni.c

```diff
@@ -5,105 +5,91 @@
 */
 #include "all.h"
 
-
 /* functions
 */
-  u3_noun
-  u3qdb_uni(u3_noun a,
-            u3_noun b)
-  {
-    if ( u3_nul == a ) {
-      return u3k(b);
-    }
-    else if ( u3_nul == b ) {
-      return u3k(a);
-    }
-    else {
-      u3_noun l_a, n_a, r_a, lr_a, p_n_a, q_n_a;
-      u3_noun l_b, n_b, r_b, lr_b, p_n_b, q_n_b;
+u3_noun
+u3qdb_uni(u3_noun a,
+          u3_noun b)
+{
+  if ( u3_nul == a ) {
+    return u3k(b);
+  }
+  else if ( u3_nul == b ) {
+    return u3k(a);
+  }
+  else {
+    u3_noun l_a, n_a, r_a, lr_a, p_n_a, q_n_a;
+    u3_noun l_b, n_b, r_b, lr_b, p_n_b, q_n_b;
+    u3x_cell(a, &n_a, &lr_a);
+    u3x_cell(b, &n_b, &lr_b);
+    u3x_cell(n_a, &p_n_a, &q_n_a);
+    u3x_cell(n_b, &p_n_b, &q_n_b);
+    u3x_cell(lr_a, &l_a, &r_a);
```

Comment:
Is there not a u3x_trel() for this purpose?

Reply:
Indeed there is. This is one of those unused jets, so I made the minimal stylistic change and moved on. But it seems reasonable to go a little further.

Reply:
Fixed